From c2506f6fa43ecd2da3208dfe8c4b3058ddb7d73c Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:39:11 -0400
Subject: [ethernet] i40e: clear VEB stats when pf stats are cleared

Message-id: <1409658042-22689-191-git-send-email-sassmann@redhat.com>
Patchwork-id: 90385
O-Subject: [RHEL7.1 PATCH 190/281] i40e: clear VEB stats when pf stats are cleared
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

The VEB really is part of the whole PF and should be cleared
at the same time.

Change-ID: Ia1d4d1df5cf421f2578a22486650dd256cc4617a
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit e91fdf7666fa4c6a5111f8b20304a990bc77e89e)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index ebf40fc..25672b3 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -444,9 +444,21 @@ void i40e_vsi_reset_stats(struct i40e_vsi *vsi)
  **/
 void i40e_pf_reset_stats(struct i40e_pf *pf)
 {
+ int i;
+
  memset(&pf->stats, 0, sizeof(pf->stats));
  memset(&pf->stats_offsets, 0, sizeof(pf->stats_offsets));
  pf->stat_offsets_loaded = false;
+
+ for (i = 0; i < I40E_MAX_VEB; i++) {
+  if (pf->veb[i]) {
+   memset(&pf->veb[i]->stats, 0,
+          sizeof(pf->veb[i]->stats));
+   memset(&pf->veb[i]->stats_offsets, 0,
+          sizeof(pf->veb[i]->stats_offsets));
+   pf->veb[i]->stat_offsets_loaded = false;
+  }
+ }
 }
 
 /**
-- 
1.7.1